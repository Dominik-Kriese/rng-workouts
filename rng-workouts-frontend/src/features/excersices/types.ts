export const muscleGroups = <const>[
    'Chest',
    'Shoulder',
    'Biceps',
    'Triceps',
    'Upper Back',
    'Latissimus',
    'Quadriceps',
    'Abs',
    'Hamstrings'
]

export type MuscleGroup = typeof muscleGroups[number]

export type Exercise = {
    name: string;
    muscleGroups: MuscleGroup[];
}
