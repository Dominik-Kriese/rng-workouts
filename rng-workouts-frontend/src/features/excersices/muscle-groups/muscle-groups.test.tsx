import React from 'react';
import {shallow} from 'enzyme';
import 'jest-styled-components';
import {MuscleGroup} from '../types';
import MuscleGroups from './muscle-groups';

describe('Muscle Groups', () => {
   const muscleGroups: MuscleGroup[] = [
       'Abs', 'Biceps'
   ];

   it('displays the muscle groups', () => {
      const groups = shallow(<MuscleGroups muscleGroups={muscleGroups} />)
          .dive()
          .find('p');
      expect(groups).toMatchSnapshot();
   });
});
