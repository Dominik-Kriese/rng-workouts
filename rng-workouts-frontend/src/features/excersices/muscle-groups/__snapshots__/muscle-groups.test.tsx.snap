// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Muscle Groups displays the muscle groups 1`] = `
.c0 {
  font-size: 0.75rem;
  opacity: 0.6;
}

<p
  className="c0"
>
  Abs, Biceps
</p>
`;
